source/read/cursors.txt

```diff
@@ -0,0 +1,145 @@
+-------------------------------
+
+Retrieve documents from a cursor individually by calling the ``mongoc_cursor_next()`` function.
+When calling the ``mongoc_cursor_next()`` function, the function will iterate the cursor and
+set the ``bson`` parameter to the next document in the cursor.
+
+The following example finds all documents in a collection with a ``name`` value
+of ``"Dunkin' Donuts"``. It then prints the first document in the cursor by calling the
+``mongoc_cursor_next()`` function.
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/cursors.c
+      :start-after: start-cursor-next
+      :end-before: end-cursor-next
+      :language: c
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      { "_id" : { "$oid" : "..." }, "address" : { "building" : "87-77", "coord" : [ { "$numberDouble" : "-73.802179999999992788" }, { "$numberDouble" : "40.707304000000000599" } ], "street" : "Parsons Boulevard", "zipcode" : "11432" }, "borough" : "Queens", "cuisine" : "Donuts", "grades" : [ { "date" : { "$date" : { "$numberLong" : "1409616000000" } }, "grade" : "A", "score" : { "$numberInt" : "12" } }, { "date" : { "$date" : { "$numberLong" : "1377561600000" } }, "grade" : "A", "score" : { "$numberInt" : "10" } }, { "date" : { "$date" : { "$numberLong" : "1365638400000" } }, "grade" : "A", "score" : { "$numberInt" : "4" } }, { "date" : { "$date" : { "$numberLong" : "1332460800000" } }, "grade" : "A", "score" : { "$numberInt" : "9" } } ], "name" : "Dunkin' Donuts", "restaurant_id" : "40392410" }
+
+Close a Cursor
+--------------
+
+To close a cursor and release all associated resources, call the
+``mongoc_cursor_destroy()`` function as shown in the following example:
+
+.. literalinclude:: /includes/read/cursors.c
+   :start-after: start-cursor-close
+   :end-before: end-cursor-close
+   :language: c
+   :dedent:
+
+Tailable Cursors
+----------------
+
+When querying on a :manual:`capped collection </core/capped-collections/>`, you
+can use a **tailable cursor** that remains open after the client exhausts the
+results in a cursor. To create a tailable cursor on a capped collection,
+specify the ``tailable`` and ``awaitData`` options when performing a find operation.
+
+The following example creates a tailable cursor on a capped collection:
+
+.. literalinclude:: /includes/read/cursors.c
+   :start-after: start-tailable-cursor
+   :end-before: end-tailable-cursor
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about tailable cursors and their usage, see the :manual:`Tailable Cursors guide
+</core/tailable-cursors/>` in the {+mdb-server+} manual.
+
+Troubleshooting
+---------------
+
+"*CursorNotFound* cursor id not valid at server"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cursors in MongoDB can timeout on the server if they've been open for
+a long time without any operations being performed on them. This can
+lead to a ``CursorNotFound`` exception when you try to iterate through the cursor.
```

Comment:
S: could you add another quick sentence about how to troubleshoot this error (I'm guessing just rerun the operation and return a new cursor)?